File: src/anry/project/Introduction/WhatAnArray.py
class DVD:
    def __init__(self, name, release_year, director):
        self.name = name
        self.release_year = release_year
        self.director = director

    def __str__(self):
        return "{}, director by {}, release in {}".format(
            self.name,
            self.director,
            self.release_year
        )


if __name__ == '__main__':
    # init an array
    lst_dvd_collections = [None] * 15

    # write some item to array
    avengersDVD = DVD("The Avengers", 2012, "Joss Whedon")
    incrediblesDVD = DVD("The Incredibles", 2004, "Brad Bird")
    findingDoryDVD = DVD("Finding Dory", 2016, "Andrew Stanton")
    lionKingDVD = DVD("The Lion King", 2019, "Jon Favreau")
    lst_dvd_collections[7] = avengersDVD
    lst_dvd_collections[3] = incrediblesDVD
    lst_dvd_collections[9] = findingDoryDVD
    lst_dvd_collections[2] = lionKingDVD

    # replace item in array by other array
    starWarsDVD = DVD("Star Wars", 1977, "George Lucas")
    lst_dvd_collections[3] = starWarsDVD

    # read item from array
    print(lst_dvd_collections[7])
    print(lst_dvd_collections[10])
    print(lst_dvd_collections[3])

    # write item into array with a loop
    lst_square_number = [0] * 10
    for i in range(10):
        lst_square_number[i] = (i + 1) * (i + 1)

    # read item from array with a loop
    for i in range(10):
        print(lst_square_number[i])

File: src/anry/project/Chap01_Introduction/WhatAnArray.java
package anry.project.Chap01_Introduction;

public class WhatAnArray {
  public static class DVD {
    public String name;
    public int releaseYear;
    public String director;

    public DVD(String name, int releaseYear, String director) {
      this.name = name;
      this.releaseYear = releaseYear;
      this.director = director;
    }

    @Override
    public String toString() {
      return this.name + ", director by " + this.director + ", release in " + this.releaseYear;
    }
  }


  /**
   *
   */
  public static void main(String[] args) {
    // init an array
    DVD[] dvdCollections = new DVD[15];

    // write some item to array
    DVD avengersDVD = new DVD("The Avengers", 2012, "Joss Whedon");
    DVD incrediblesDVD = new DVD("The Incredibles", 2004, "Brad Bird");
    DVD findingDoryDVD = new DVD("Finding Dory", 2016, "Andrew Stanton");
    DVD lionKingDVD = new DVD("The Lion King", 2019, "Jon Favreau");
    dvdCollections[7] = avengersDVD;
    dvdCollections[3] = incrediblesDVD;
    dvdCollections[9] = findingDoryDVD;
    dvdCollections[2] = lionKingDVD;

    // replace item in array by other array
    DVD starWarsDVD = new DVD("Star Wars", 1977, "George Lucas");
    dvdCollections[3] = starWarsDVD;

    // read item from array
    System.out.println(dvdCollections[7]);
    System.out.println(dvdCollections[10]);
    System.out.println(dvdCollections[3]);

    // write item into array with a loop
    int[] squareNumber = new int[10];
    for (int i = 0; i < 10; i ++) {
      squareNumber[i] = (i + 1) * (i + 1);
    }

    // read item from array with a loop
    for (int i = 0 ; i < 10 ; i ++) {
      System.out.println(squareNumber[i]);
    }
  }

}
